### PR TITLE
Pytorch Lightning Example

### PR DESCRIPTION
**Pull request by tchaton**
This PR adds an example on how to use Pytorch Lightning with Pytorch Geometric.

### CODE REVIEW

File: examples/pytorch_lightning/cora_dna.py [hunk 1]
```diff
@@ -0,0 +1,370 @@
...
+    """
+
+    NAME = "cora"
+
+    def __init__(self,
+                 num_workers: int = 1,
+                 batch_size: int = 8,
+                 drop_last: bool = True,
+                 pin_memory: bool = True,
+                 num_layers: int = None):
```
Comment (Borda) [inline examples/pytorch_lightning/cora_dna.py:79]:
```suggestion,
                 num_layers: int
                 num_workers: int = 1,
                 batch_size: int = 8,
                 drop_last: bool = True,
                 pin_memory: bool = True):
```
does not make sense to have it optional if the default value is not feasible...

File: examples/pytorch_lightning/cora_dna.py [hunk 2]
```diff
@@ -0,0 +1,370 @@
...
+                 groups: int = 16,
+                 dropout: float = 0.8,
+                 cached: bool = False,
+                 num_features: int = None,
+                 num_classes: int = None,
+                 ):
+        super().__init__()
+
+        assert num_features is not None
+        assert num_classes is not None
```
Comment (Borda) [inline examples/pytorch_lightning/cora_dna.py:187]: the same here, do not make them then default as None

**Comment by tchaton** [comment]
Hey @rusty1s,

Any update on this PR ? We are waiting for it to be merged to delete it on your side ?

Best regards,
chaton

**Comment by rusty1s** [comment]
Will try to merge by the end of the week. Sorry for the delay!

File: examples/pytorch_lightning/gin.py [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+    seed_everything(42)
+    datamodule = IMDBBinary()
+    model = GIN(datamodule.num_features, datamodule.num_classes)
+    checkpoint_callback = ModelCheckpoint(monitor='val_acc')
+    trainer = Trainer(gpus=1, max_epochs=20, callbacks=[checkpoint_callback])
+    trainer.fit(model, datamodule=datamodule)
+    print('--------------------')
+    print(trainer.callbacks[0].best_model_path)
+    print('--------------------')
+    model = GIN.load_from_checkpoint(checkpoint_callback.best_model_path)
```
Comment (tchaton) [inline examples/pytorch_lightning/gin.py:127]: You don't need this, test should do it automatically.

File: examples/pytorch_lightning/gin.py [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+
+def main():
+    seed_everything(42)
+    datamodule = IMDBBinary()
+    model = GIN(datamodule.num_features, datamodule.num_classes)
+    checkpoint_callback = ModelCheckpoint(monitor='val_acc')
+    trainer = Trainer(gpus=1, max_epochs=20, callbacks=[checkpoint_callback])
+    trainer.fit(model, datamodule=datamodule)
+    print('--------------------')
+    print(trainer.callbacks[0].best_model_path)
```
Comment (tchaton) [inline examples/pytorch_lightning/gin.py:125]: What do you get there ?

**Pull request merged**